data_structures/MIddleElement_S.ll.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file 
+ * @brief Implementation of [Finding middle element in 
+ * a single linked list](https://simple.wikipedia.org/wiki/Linked_list)
+ * @details
+ * The linked list is a data structure used for holding a sequence of
+ * values, which can be added, displayed, reversed, or removed.
+ * ### Algorithm
+ * This can be done by the slow and fast pointer approach 
+ * (two pointer approach),basically in this we make two pointer
+ * slow and fast we point slow = head , and fast to the head->next
+ * and if length of the L.L is even then traverse till fast!=NULL
+ * and we move slow by one step and fast by one step and when fast
+ * became null then we just print the slow->data , (our mid element).
+ * For the odd length all the process will be same but only one diff. 
+ * isthat we need to check instead of fast!=null we have to chek 
+ * fast->next!=null and similarlly we print the slow->data
+ */
+
+ #include <iostream>
+
+/**
+ * A Node class containing a value and pointer to another link
+ */
+class node{
+public:
+    int data;
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c++
int64_t data = 0;       ///< The value/key of the node.
```